beacon-chain/blockchain/kzg/validation.go

```diff
@@ -27,6 +45,92 @@ func Verify(sidecars ...blocks.ROBlob) error {
 	return kzgContext.VerifyBlobKZGProofBatch(blobs, cmts, proofs)
 }
 
+func BlobToKZGCommitment(blob *Blob) (Commitment, error) {
+	comm, err := kzg4844.BlobToCommitment(kzg4844.Blob(*blob))
+	if err != nil {
+		return Commitment{}, err
+	}
+	return Commitment(comm), nil
+}
+
+func ComputeBlobKZGProof(blob *Blob, commitment Commitment) (Proof, error) {
+	proof, err := kzg4844.ComputeBlobProof(kzg4844.Blob(*blob), kzg4844.Commitment(commitment))
+	if err != nil {
+		return [48]byte{}, err
+	}
+	return Proof(proof), nil
+}
+
+func ComputeCellsAndKZGProofs(blob *Blob) ([ckzg4844.CellsPerExtBlob]Cell, [ckzg4844.CellsPerExtBlob]Proof, error) {
+	ckzgBlob := ckzg4844.Blob(*blob)
+	_cells, _proofs, err := ckzg4844.ComputeCellsAndKZGProofs(&ckzgBlob)
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, [ckzg4844.CellsPerExtBlob]Proof{}, err
+	}
+
+	// Convert Cells and Proofs to types defined in this package
+	var cells [ckzg4844.CellsPerExtBlob]Cell
+	for i, cell := range _cells {
+		cells[i] = Cell(cell)
+	}
+
+	var proofs [ckzg4844.CellsPerExtBlob]Proof
+	for i, proof := range _proofs {
+		proofs[i] = Proof(proof)
+	}
+
+	return cells, proofs, nil
+}
+
+func VerifyCellKZGProof(commitmentBytes Bytes48, cellId uint64, cell Cell, proofBytes Bytes48) (bool, error) {
+	return ckzg4844.VerifyCellKZGProof(commitmentBytes, cellId, ckzg4844.Cell(cell), proofBytes)
+}
+
+func VerifyCellKZGProofBatch(commitmentsBytes []Bytes48, rowIndices, columnIndices []uint64, cells []Cell, proofsBytes []Bytes48) (bool, error) {
+	// Convert Cell type to ckgz4844.Cell
+	var ckzgCells []ckzg4844.Cell
+	for i, cell := range cells {
+		ckzgCells[i] = ckzg4844.Cell(cell)
+	}
+
+	return ckzg4844.VerifyCellKZGProofBatch(commitmentsBytes, rowIndices, columnIndices, ckzgCells, proofsBytes)
+}
+
+func RecoverAllCells(cellIds []uint64, cells []Cell) ([ckzg4844.CellsPerExtBlob]Cell, error) {
+	// Convert Cell type to ckgz4844.Cell
+	var ckzgCells []ckzg4844.Cell
+	for i, cell := range cells {
+		ckzgCells[i] = ckzg4844.Cell(cell)
+	}
+
+	recoveredCells, err := ckzg4844.RecoverAllCells(cellIds, ckzgCells[:])
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, err
+	}
+
+	// Convert ckzg cells to `Cell` used in API
+	var ret [ckzg4844.CellsPerExtBlob]Cell
+	for i, cell := range recoveredCells {
+		ret[i] = Cell(cell)
+	}
+	return ret, nil
+}
+
+func CellsToBlob(cells [ckzg4844.CellsPerExtBlob]Cell) (Blob, error) {
+	// Convert Cell type to ckgz4844.Cell
+	var ckzgCells [ckzg4844.CellsPerExtBlob]ckzg4844.Cell
+	for i, cell := range cells {
```

Comment:
Using 
```go
for i, cell := range cells {
    ...
}
```

creates a copy of cell.

Cell is quite big.

Could be replaced by something like:

```go
for i := range cells {
		ckzgCells[i] = ckzg4844.Cell(cell[i])
	}
```